### PR TITLE
ci: setup prima

### PR DESCRIPTION
**Pull request by Pranavchiku**
Branch: https://github.com/Pranavchiku/prima/tree/lf13
commit: https://github.com/Pranavchiku/prima/commit/b3f5b55c94b3b0b2568fe9af97a8e34a48cac7b4

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -499,6 +499,17 @@ jobs:
...
             ./build1.sh
 
+      - name: Test PRIMA
+        shell: bash -e -l {0}
+        run: |
+            git clone https://github.com/Pranavchiku/prima.git
+            cd prima
+            git checkout -t origin/lf13
+            git checkout b3f5b55c94b3b0b2568fe9af97a8e34a48cac7b4
+            FC="$(pwd)/../src/bin/lfortran --cpp --skip-pass=pass_array_by_data"
```
Comment (gxyd) [inline .github/workflows/CI.yml:509]: Odd that this command is raising an error in CI, while locally this works for me

Comment (gxyd) [reply .github/workflows/CI.yml:509]: that's the case on macOS, while on ubuntu this seems to pass as expect but errors on the next command (as expected I believe)

**Comment by certik** [comment]
It works on macOS LLVM 11. It fails in our LLVM backend for LLVM 19:
```
[ 82%] Building Fortran object fortran/CMakeFiles/example_bobyqa_fortran_1_exe.dir/examples/bobyqa/bobyqa_example_1.f90.o
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/runner/work/lfortran/lfortran/src/bin/lfortran.cpp", line 2723
    LCompilers::print_stack_on_segfault();
  File "/Users/runner/work/lfortran/lfortran/src/bin/lfortran.cpp", line 2620
    return compile_src_to_object_file(arg_file, outfile, false,
  File "/Users/runner/work/lfortran/lfortran/src/bin/lfortran.cpp", line 1070
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/Users/runner/work/lfortran/lfortran/src/lfortran/fortran_evaluator.cpp", line 382
    compiler_options, run_fn, infile);
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 10394
    v.visit_asr((ASR::asr_t&)asr);
  File "src/libasr/../libasr/asr_base_visitor.h", line 316
  File "src/libasr/../libasr/asr_base_visitor.h", line 292
  File "src/libasr/../libasr/asr_base_visitor.h", line 317
  File "src/libasr/../libasr/asr_base_visitor.h", line 22
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 974
    ASR::symbol_t *mod = x.m_symtab->get_symbol(item);
  File "src/libasr/../libasr/asr_base_visitor.h", line 319
  File "src/libasr/../libasr/asr_base_visitor.h", line 30
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 3173
    finish_module_init_function_prototype(x);
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4298
    ASR::Function_t *s = ASR::down_cast<ASR::Function_t>(item.second);
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4054
    visit_procedures(x);
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4251
    this->visit_stmt(*x.m_body[i]);
  File "src/libasr/../libasr/asr_base_visitor.h", line 336
  File "src/libasr/../libasr/asr_base_visitor.h", line 55
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5278
    value_data = llvm_utils->CreateLoad(arr_descr->get_pointer_to_data(value));
  File "/Users/runner/work/lfortran/lfortran/src/libasr/codegen/llvm_utils.cpp", line 1656
    LCOMPILERS_ASSERT(type);
AssertFailed: type
```
On linux / LLVM 11 it builds the binary but the binary segfaults:
```
/home/runner/work/_temp/e0d9e13d-d530-46c7-99f1-23911af21ddc.sh: line 6:  4957 Segmentation fault      (core dumped) ./build/fortran/example_bobyqa_fortran_1_exe
```
And finally linux / LLVM 19 fails with:
```
LCOMPILERS_ASSERT failed: /home/runner/work/lfortran/lfortran/src/libasr/codegen/llvm_utils.cpp
function CreateLoad(), line number 1656 at 
type
Aborted (core dumped)
```

Which is the same error as macOS / LLVM 19.


Why not enable this for macOS LLVM 11 only for now, that way everything passes? And we'll fix up linux as we go, and then we have to isolate the bug for LLVM 19 and fix it also.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Pranavchiku) [reply .github/workflows/CI.yml]: Yes, it should have worked but I suspect we are expecting `FC` in Makefile. That is fine, pushed a commit to fix this.

**Review by certik — APPROVED** [review]

**Pull request merged**